www/src/pages/en/usage/next-auth.md

```diff
@@ -56,11 +65,76 @@ declare module "next-auth" {
 }
 ```
 
+The same pattern can be used to add any other data to the `session` object, such as a `role` field, but **should not be misused to store sensitive data** on the client.
+
+### Usage with tRPC
+
+When using NextAuth.js with tRPC, you can create reusable, protected procedures using [middleware](https://trpc.io/docs/v10/middlewares). This allows you to create procedures that can only be accessed by authenticated users. `create-t3-app` sets all of this up for you, allowing you to easily access the session object within authenticated procedures.
+
+This is done in a two step process:
+
+1. Grab the session from the request headers using the [unstable_getServerSession](https://next-auth.js.org/configuration/nextjs#unstable_getserversession) function (don't worry, its not unstable cause of security reasons, but cause of its API might change). The advantage of using `unstable_getServerSession` instead of the regular `getSession` is that its a server-side only function and doesn't trigger unnecessary fetch calls. `create-t3-app` creates a helper function that abstacts this weird, funky API away.
```

Comment:
"that abstacts this weird, funky API away."

it's late. words are hard

Reply:
maybe peculiar